backend/source/apis/friends/get_friends.go

```diff
@@ -0,0 +1,69 @@
+package friends
+
+import (
+	"2021-fall-cs160-team-Mochi/backend/source/apis/commonutils"
+	"2021-fall-cs160-team-Mochi/backend/source/generated/models"
+	"2021-fall-cs160-team-Mochi/backend/source/generated/restapi/operations/friends_v1"
+	"net/http"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/jinzhu/gorm"
+)
+
+func GetFriendsV1Handler(db *gorm.DB) friends_v1.GetFriendsV1HandlerFunc {
+	return func(params friends_v1.GetFriendsV1Params) (responder middleware.Responder) {
+		getFriendObject, errResp := processGetfriendsRequest(db, params)
+		if errResp != nil {
+			switch errResp.StatusCode {
+			case http.StatusBadRequest:
+				return friends_v1.NewGetFriendsV1BadRequest().WithPayload(errResp)
+			case http.StatusUnauthorized:
+				return friends_v1.NewGetFriendsV1Unauthorized().WithPayload(errResp)
+			case http.StatusForbidden:
+				return friends_v1.NewGetFriendsV1Forbidden().WithPayload(errResp)
+			case http.StatusNotFound:
+				return friends_v1.NewGetFriendsV1NotFound().WithPayload(errResp)
+			case http.StatusConflict:
+				return friends_v1.NewGetFriendsV1Conflict().WithPayload(errResp)
+			default:
+				return friends_v1.NewGetFriendsV1InternalServerError().WithPayload(errResp)
+			}
+		}
+		resp := friends_v1.NewGetFriendsV1OK()
+		resp.SetPayload(getFriendObject)
+		return resp
+	}
+}
+
+func processGetfriendsRequest(db *gorm.DB, params friends_v1.GetFriendsV1Params) (resp *models.GetFriendObject, errResp *models.ErrResponse) {
+	payload, errResp := commonutils.ExtractJWT(params.HTTPRequest)
+	if errResp != nil {
+		return
+	}
+	username := payload.Username
+
+	resp = &models.GetFriendObject{
+		Friends: []*models.FriendResponse{},
+	}
+
+	rawSQL := `SELECT username2 as username FROM friends WHERE username = ?`
+
```

Comment:
Like post friends, we should do
`SELECT username2, username as username FROM friends WHERE username = ? || username2 = ?`